examples/jsm/tsl/shadows/TileShadowNode.js

```diff
@@ -0,0 +1,452 @@
+import {
+	Vector3,
+	Object3D,
+	ShadowBaseNode,
+	Plane,
+	Line3,
+	DepthArrayTexture,
+	LessCompare,
+	Vector2,
+	RedFormat,
+	NodeMaterial,
+	ArrayCamera,
+	VSMShadowMap,
+	NodeUtils,
+	RendererUtils,
+	Quaternion } from 'three/webgpu';
+
+import { min, Fn, shadow, vec4, positionWorld, renderGroup, reference, objectPosition } from 'three/tsl';
+import { TileShadowNodeHelper } from './TileShadowNodeHelper.js';
+
+const { getDataFromObject } = NodeUtils;
+const { resetRendererAndSceneState, restoreRendererAndSceneState } = RendererUtils;
+let _rendererState;
+
+const shadowMaterialLib = /*@__PURE__*/ new WeakMap();
```

Comment:
It would be nice to share `shadowMaterialLib` and its derived functions like `getShadowMaterial` of `ShadowNode`